### ISSUE TITLE
Using Iconify Vue3 Icon Component in inline element causes inline elements content to render twice

### ISSUE THREAD

**Issue by awacode21** [iconify/iconify#114]
When i use any iconify icon via its vue component within an `<a>` tag it causes the rest of the `<a>` tags content to render twice:

Usage: 
```
 <a :href="'mailto:' + footer?.email">
          <Icon icon="ic:outline-email" :inline="true"></Icon>
          <span>{{ footer?.email }}</span>
</a>
```

Result:
<img width="512" alt="Bildschirmfoto 2022-02-24 um 21 03 15" src="https://user-images.githubusercontent.com/30939926/155598815-c8015fc2-ae09-4bac-b5b7-910eae2c5508.png">

**Comment by awacode21** [comment]
This issue seems not be only related to `<a>` tags. It seems to be a problem for all inline tags:

Example:
```
<span>
          <Icon icon="ic:outline-email" :inline="true"></Icon>
          Test Content
</span>
```
<img width="517" alt="Bildschirmfoto 2022-02-24 um 21 13 03" src="https://user-images.githubusercontent.com/30939926/155600210-a5cfffa0-15ee-46bd-bcb0-71027ef3c54d.png">

<img width="228" alt="Bildschirmfoto 2022-02-24 um 21 13 41" src="https://user-images.githubusercontent.com/30939926/155600281-eb722f2c-a31c-46da-976a-b10dd4c78e15.png">

**Comment by awacode21** [comment]
Giving those inline elements `display: block;` or `display: inline-block;` does not fix the issue!!!

**Comment by cyberalien** [comment]
I've tried replicating that bug, but everything works correctly.

Icon component cannot affect content outside of it, so it doesn't make sense for component to cause text to be displayed twice. Maybe you are using some script that affects `svg` tag? Try instead of icon component use svg by copying it from browser, see bug remains. If bug remains, it is definitely not caused by icon component.

**Comment by awacode21** [comment]
It also does not seem to be related to inline elements:
```
<div>
          <Icon icon="ic:outline-email" :inline="true"></Icon>
          <span>Test Content</span>
</div>
```
<img width="532" alt="Bildschirmfoto 2022-02-24 um 21 24 27" src="https://user-images.githubusercontent.com/30939926/155601811-8e65292a-f684-4d85-b433-55a42b627403.png">

It always happens as long as there is sibling content to the Icon Component. So it only can be worked around by wrapping the icon component with a parent contain which contains no other children:
```
<div>
          <span>
            <Icon icon="ic:outline-email" :inline="false"></Icon>
          </span>
          <span>Test Content</span>
</div>
```
<img width="529" alt="Bildschirmfoto 2022-02-24 um 21 33 10" src="https://user-images.githubusercontent.com/30939926/155603071-5761e095-ebe9-4105-a420-bccd33e20910.png">
<img width="130" alt="Bildschirmfoto 2022-02-24 um 21 33 17" src="https://user-images.githubusercontent.com/30939926/155603092-71e2b5ad-ad39-4022-be28-862c442146dd.png">

**Comment by awacode21** [comment]
> I've tried replicating that bug, but everything works correctly.
> 
> Icon component cannot affect content outside of it, so it doesn't make sense for component to cause text to be displayed twice. Maybe you are using some script that affects `svg` tag? Try instead of icon component use svg by copying it from browser, see bug remains. If bug remains, it is definitely not caused by icon component.

i tried what you suggested, copying the svg from browser and replace the icon component by the svg. then the issue does not appear.
But i really don't understand why that happens. I don't process anything on svg. I've just started with a nearly empty project, starting to build a footer. There is nothing complex, only requesting footer content from cms and display it.
i am using windicss. Do you think this could influence iconify somehow?

**Comment by cyberalien** [comment]
What bundler are you using and what version of bundler and Vue?

**Comment by awacode21** [comment]
```
"dependencies": {
    "@vueuse/head": "^0.7.5",
    "axios": "^0.21.4",
    "node-fetch": "^3.1.0",
    "pinia": "^2.0.11",
    "query-string": "^7.0.1",
    "storyblok-js-client": "^4.2.1",
    "uuid": "^8.3.2",
    "vue": "^3.2.25",
    "vue-router": "^4.0.12"
  },
  "devDependencies": {
    "@iconify/vue": "^3.1.3",
    "@types/node": "^17.0.8",
    "@types/uuid": "^8.3.3",
    "@typescript-eslint/eslint-plugin": "^5.10.0",
    "@typescript-eslint/parser": "^5.10.0",
    "@vitejs/plugin-vue": "^2.0.0",
    "@vue/eslint-config-typescript": "^10.0.0",
    "compression": "^1.7.4",
    "cross-env": "^7.0.3",
    "dotenv": "^15.0.0",
    "eslint": "^8.7.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-vue": "^8.3.0",
    "express": "^4.17.2",
    "lint-staged": "^12.2.0",
    "prettier": "2.5.1",
    "sass": "^1.49.8",
    "serialize-javascript": "^6.0.0",
    "serve-static": "^1.14.2",
    "typescript": "^4.4.4",
    "vite": "^2.7.2",
    "vite-plugin-eslint": "^1.3.0",
    "vite-plugin-windicss": "^1.7.1",
    "vue-tsc": "^0.29.8",
    "windicss": "^3.4.3"
  }
```

**Comment by awacode21** [comment]
Could it be a problem that i am using Server Side Rendering?

**Comment by cyberalien** [comment]
> Could it be a problem that i am using Server Side Rendering?

Could be. I haven't actually tested Vue 3 component properly with SSR because last time I checked Nuxt, wasn't available for Vue 3. For SSR I've only tested and used Svelte and React components, which are pretty much identical to Vue except for framework specific code.

But it is available now, so I'll test it.

**Comment by awacode21** [comment]
Just found this about Nuxt.js and SSR. I am not using Nuxt. But i guess the problem is the same for vue3.
Icon component does get its icon on mounted, so it cannot be rendered on server. Seems that for SSR i need to use an offline bundle:
<img width="769" alt="Bildschirmfoto 2022-02-24 um 21 54 43" src="https://user-images.githubusercontent.com/30939926/155606186-7344f7f2-8475-4496-a688-7a75037acce9.png">

**Comment by cyberalien** [comment]
Tested with Nuxt and I can replicate that bug! Very weird bug. It is great news because now I can work on fixing it

**Comment by awacode21** [comment]
Thank you very much that you investigate time on this issue. I guess the problem is that icon is loaded onMounted. but for SSR it needs to be available onCreated/setup

**Comment by cyberalien** [comment]
Unfortunately that's not it. Even when replacing event with `onCreate` and rendering icon immediately, bug still remains.

**Comment by cyberalien** [comment]
It appeared to be very tricky bug. Bug is triggered by re-rendering icon after icon data has been loaded. Changing rendering to `onCreate` fixes issue for icons when data is instantly available, but not for dynamically loaded icons.

This will take a while to debug. It is close to midnight for me, so will do that tomorrow.

**Comment by awacode21** [comment]
Thank you very, very much

**Comment by cyberalien** [comment]
Found a solution. Issue was caused by bug in Nuxt, which I've reported https://github.com/nuxt/framework/issues/3389

There is a simple workaround, which seem to work very well, but then unit tests aren't passing. I've published update with `dev` tag, which you can install using `npm install @iconify/vue@dev`. I'll need to find a solution for unit tests before publishing it as stable version.

**Comment by danielroe** [comment]
This isn't a Nuxt bug. Changing the component state before mounting will trigger a hydration error on any server-rendered app: https://stackblitz.com/edit/vitejs-vite-ahtqpq

**Comment by cyberalien** [comment]
> This isn't a Nuxt bug. Changing the component state before mounting will trigger a hydration error on any server-rendered app: https://stackblitz.com/edit/vitejs-vite-ahtqpq

It works correctly without SSR though. I don't see anything in Vue documentation that states that changing state is not allowed in `beforeMount`

**Comment by danielroe** [comment]
Check this out: https://vuejs.org/guide/scaling-up/ssr.html#writing-ssr-friendly-code

**Comment by cyberalien** [comment]
It is not about SSR friendly code, component is meant to render nothing in SSR, but render actual data when rendering in browser.

Re-render should happen in browser after component was mounted, but it doesn't. If even is changed from `beforeMount()` to `mounted()`, it does work as expected.

**Comment by danielroe** [comment]
It works fine without SSR because there's no concept of hydration then. But hydration always attempts to match up what _would_ be rendered with what _was_ rendered. Changing the component state before mount (for example, in `beforeMount`) will result in hydration errors.

**Comment by cyberalien** [comment]
> It works fine without SSR because there's no concept of hydration then. But hydration always attempts to match up what _would_ be rendered with what _was_ rendered. Changing the component state before mount (for example, in `beforeMount`) will result in hydration errors.

I think you misunderstood. Something about SSR causes component to not trigger re-render in browser, not on server. Issue is about even not working in browser.

Without Nuxt: component is created, first render() is called that returns empty content, beforeMount() is called, it changes state, render() is called again, UI updates.

With Nuxt: component is rendered on server (empty render) as expected. Then in browser hydration happens, then beforeMount() is called, it changes state... and the rest doesn't happen.

**Comment by danielroe** [comment]
Did you check [my reproduction link](https://stackblitz.com/edit/vitejs-vite-ahtqpq)? It's not using Nuxt. If you feel this is a bug, you should open an issue at https://github.com/vuejs/core, as it is Vue behaviour you are describing.

**Comment by awacode21** [comment]
As i reported the issue and i am cleary not using nuxt (i am using vue 3 with vite, but no nuxt), it is cleary no nuxt issue, it's a general one with the SSR.

**Comment by cyberalien** [comment]
Sorry for confusion and thanks a lot for better reproduction repo!

**Comment by danielroe** [comment]
No problem at all - glad to be of help 😊

**Comment by cyberalien** [comment]
Published stable version `3.1.4`, which fixes this issue. Also fixed similar issue in Vue 2 component.
